src/main/scala/com/kinja/soy/SoyWrites.scala

```diff
@@ -21,6 +21,25 @@ trait SoyWrites[-T] extends DefaultSoyWrites {
 object SoyWrites extends DefaultSoyWrites
 
 /**
+ * The class of types that can be converted into a SoyMap. Every SoyMapWrites implies a SoyWrites.
+ */
+trait SoyMapWrites[-T] extends SoyWrites[T] {
+  def toSoy(t: T): SoyMap
+  final def and[U](w: SoyMapWrites[U]): SoyMapWrites[(T, U)] = {
+    val _toSoy = toSoy _
```

Comment:
Ah, I see. You could also add a self-type variable to the top SoyMapWrites and get around that. IE

`trait SoyMapWrites[-T] { a: =>`

and then 

`a.toSoy`